src_c/circle.c

```diff
@@ -261,6 +261,61 @@ pg_circle_collidepoint(pgCircleObject *self, PyObject *const *args,
     return RAISE(PyExc_TypeError,
                  "collidepoint requires a point or PointLike object");
 }
+static PyObject *
+pg_circle_colliderect(pgCircleObject *self, PyObject *const *args,
+                      Py_ssize_t nargs)
+{
+    SDL_Rect *tmp, temp;
+    if (nargs == 1) {
+        if (!(tmp = pgRect_FromObject(args[0], &temp))) {
+            if (PyErr_Occurred())
+                return NULL;
+            else
+                return RAISE(PyExc_TypeError,
+                             "Invalid rect, all 4 fields must be numeric");
+        }
+        return PyBool_FromLong(pgCollision_RectCircle(tmp, &(self->circle)));
+    }
+    else if (nargs == 2) {
+        if (!pg_TwoIntsFromObj(args[0], &temp.x, &temp.y) ||
+            !pg_TwoIntsFromObj(args[1], &temp.w, &temp.h)) {
+            return RAISE(PyExc_TypeError,
+                         "Invalid rect, all 4 fields must be numeric");
+        }
+    }
+    else if (nargs == 4) {
+        if (!pg_IntFromObj(args[0], &temp.x) ||
+            !pg_IntFromObj(args[1], &temp.y) ||
+            !pg_IntFromObj(args[2], &temp.w) ||
+            !pg_IntFromObj(args[3], &temp.h)) {
+            return RAISE(PyExc_TypeError,
+                         "Invalid rect, all 4 fields must be numeric");
+        }
+    }
+    else {
+        return RAISE(PyExc_TypeError,
+                     "Invalid argumets number, must be 1, 2 or 4");
+    }
+
+    return PyBool_FromLong(pgCollision_RectCircle(&temp, &(self->circle)));
+}
+
+static PyObject *
+pg_circle_collideswith(pgCircleObject *self, PyObject *arg)
+{
+    int result = 0;
+    if (pgCircle_Check(arg)) {
+        result =
+            pgCollision_CircleCircle(&self->circle, &pgCircle_AsCircle(arg));
+    }
+    else if (pgRect_Check(arg)) {
+        result = pgCollision_RectCircle(&(pgRect_AsRect(arg)), &self->circle);
+    }
+    else if (PySequence_Check(arg)) {
+        return pg_circle_collidepoint(self, (PyObject *const *)(&arg), 1);
```

Comment:
> `(PyObject *const *)`

This looks strange... @Starbuck5 can you check this?

Reply:
It's just a little hack to not have to add a new function just for this one to work

Reply:
And it works i've tested it visually

Reply:
nvm removed it